apps/meteor/server/models/raw/LivechatVisitors.ts

```diff
@@ -158,11 +159,13 @@ export class LivechatVisitorsRaw extends BaseRaw<ILivechatVisitor> implements IL
 	/**
 	 * Find visitors by their email or phone or username or name
 	 */
-	findPaginatedVisitorsByEmailOrPhoneOrNameOrUsername(
+	async findPaginatedVisitorsByEmailOrPhoneOrNameOrUsernameOrCustomField(
 		emailOrPhone: string,
 		nameOrUsername: RegExp,
 		options: FindOptions<ILivechatVisitor>,
-	): FindPaginated<FindCursor<ILivechatVisitor>> {
+	): Promise<FindPaginated<FindCursor<ILivechatVisitor>>> {
+		const allowedCF = LivechatCustomField.find({ scope: 'visitor', searchable: true }, { projection: { _id: 1 } }).map(({ _id }) => _id);
```

Comment:
ideally this method should receive the list of custom fields to search for, removing the dependency of the model